src/urllib3/_collections.py

```diff
@@ -255,7 +255,7 @@ def __init__(self, headers: Optional[ValidHTTPHeaderSource] = None, **kwargs: st
         if kwargs:
             self.extend(kwargs)
 
-    def __setitem__(self, key: str, val: str) -> None:
+    def __setitem__(self, key: Union[bytes, str], val: str) -> None:
```

Comment:
So this is true but I'm not sure if we should recommend it? Wasn't there some discussion about `bytes` headers somewhere, can't find it now.

Reply:
Is the preference to?

1. Add ignore comments to all uses
2. Only handle str, update tests, implementation and the type signature?

Reply:
I went with option 2 in #2397 but am very open to discussing other options.

Reply:
I have reverted the changes in `src/urllib3/_collections.py`. They now continue to be typed for `str` only even though the implementation allows bytes. I have added `# type: ignore` comments to the necessary tests with a comment. Thanks for clarifying.

Reply:
I appreciate you taking initiative here, we may end up deprecating using `bytes` for this case to push a little harder towards using our APIs as intended. We can handle that separately though.